scripts/run_backend_tests_test.py

```diff
@@ -916,8 +930,10 @@ def mock_subprocess_run(cmd: List[str], **_: str) -> MockProcessOutput:
             returned_output, coverage = run_backend_tests.check_coverage(
                 False, data_file=data_file)
 
-        self.assertEqual(returned_output, coverage_report_output)
-        self.assertEqual(coverage, 86)
+        expected_output = coverage_report_output.strip()
+        returned_res = returned_output.strip()
+        self.assertEqual(expected_output, returned_res)
+        self.assertEqual(coverage, 62.0)
```

Comment:
I think there should be at least one unit test that verifies the log that is displayed for the end user (the developer in this case). Maybe two tests for the cases "there is a coverage gap to report" and "there is no coverage gap to report".

Reply:
@seanlip does coverage gap mean the condition when all the files will have complete coverage?

Reply:
A "coverage gap" means that some file has a lack of coverage.

Reply:
@seanlip I have one last question. The test case I added currently focuses on scenarios where there is a coverage gap (files lack coverage). I believe we can enhance it by combining it with another test case, where some files have complete coverage and others have a lack of coverage. This way, we can test both scenarios within a single test case. Does this approach sound correct?

Reply:
I suggest you have one test where there's a coverage gap, one test where there's no coverage gap, and one test that mixes the two. 

It is good to have focused test cases since that will make it easier to identify when an issue occurs. With a single test case that covers multiple scenarios, diagnosing issues becomes harder.

Reply:
@seanlip I have included tests for all three conditions in first there are only uncovered files, second is then there are only covered lines and third is when both covered and uncovered files will be present in coverage report. Thanks.